docs: Javadocs no longer build

### What happened?

This is because Arrow no longer builds the Arrow Java docs which we link to.

### Stack Trace

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.12.1:site (default-site) on project arrow-adbc-java-root: Error generating maven-javadoc-plugin:3.11.1:aggregate report: 
Error:  Exit code: 1
Error: [ERROR] error: Error fetching URL: https://arrow.apache.org/docs/java/reference/ (java.io.FileNotFoundException: https://arrow.apache.org/docs/java/reference/package-list)
Error: [ERROR] 1 error
Error:  Command line was: /usr/lib/jvm/temurin-17-jdk-amd64/bin/javadoc -J-Duser.language= -J-Duser.country= @options @packages
Error:  
Error:  Refer to the generated Javadoc files in '/home/runner/work/arrow-adbc/arrow-adbc/java/target/site/apidocs' dir.
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error: Process completed with exit code 1.
```

### How can we reproduce the bug?

_No response_

### Environment/Setup

_No response_

--- a/ci/scripts/docs_build.sh
+++ b/ci/scripts/docs_build.sh
@@ -26,7 +26,7 @@ main() {
     popd
 
     pushd "$source_dir/java"
-    mvn site
+    mvn --no-transfer-progress site
     popd
 
     pushd "$source_dir/docs"

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -239,9 +239,10 @@
         <configuration>
           <!-- Ignore missing @return, @throws -->
           <doclint>all,-missing</doclint>
-          <links>
-            <link>https://arrow.apache.org/docs/java/reference/</link>
-          </links>
+          <!-- TODO(apache/arrow-adbc#2454): upstream stopped building this -->
+          <!-- <links> -->
+          <!--   <link>https://arrow.apache.org/docs/java/reference/</link> -->
+          <!-- </links> -->
           <!-- Stop it from trying to document the tests -->
           <show>public</show>
         </configuration>

